examples/esp_idf/common/sample_credentials.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2024 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include <string.h>
+#include <golioth/client.h>
+#include "wifi.h"
+#include "nvs.h"
+
+#ifndef CONFIG_GOLIOTH_SAMPLE_HARDCODED_CREDENTIALS
+#include "nvs.h"
+#include "shell.h"
+#include "wifi.h"
+#endif
+
+#define TAG "sample_credentials"
+
+static struct golioth_client_config _golioth_client_config = {
+    .credentials =
+        {
+            .auth_type = GOLIOTH_TLS_AUTH_TYPE_PSK,
+            .psk =
+                {
+                    .psk_id = NULL,
+                    .psk_id_len = 0,
+                    .psk = NULL,
+                    .psk_len = 0,
+                },
+        },
+};
+
+const struct golioth_client_config *golioth_sample_credentials_get(void)
+{
+#ifdef CONFIG_GOLIOTH_SAMPLE_HARDCODED_CREDENTIALS
+
+    _golioth_client_config.credentials.psk.psk_id = CONFIG_GOLIOTH_SAMPLE_PSK_ID;
+    _golioth_client_config.credentials.psk.psk_id_len = strlen(CONFIG_GOLIOTH_SAMPLE_PSK_ID);
+    _golioth_client_config.credentials.psk.psk = CONFIG_GOLIOTH_SAMPLE_PSK;
+    _golioth_client_config.credentials.psk.psk_len = strlen(CONFIG_GOLIOTH_SAMPLE_PSK);
+
+    // Initialize WiFi and wait for it to connect
+    printf("calling wifi_init with %s and %s\n", CONFIG_GOLIOTH_SAMPLE_WIFI_SSID, CONFIG_GOLIOTH_SAMPLE_WIFI_PSK);
```

Comment:
We shouldn't be doing anything with WiFi in this module. It probably makes sense to commonize the bits of the sample that get WiFi credentials, but that can be in a separate ticket/PR and a separate module. We also don't necessarily want to tie the runtime vs hardcoded choice for WiFi settings to the same choice for Golioth credentials. For example, the user could select hardcoded PSK but runtime WiFi credentials (maybe unlikely, but possible).